internal/addrs/removable.go

```diff
@@ -19,6 +19,13 @@ type ConfigRemovable interface {
 	String() string
 }
 
+// DestroyableConfigRemovable wraps the ConfigRemovable and provides a new option, Destroy, that is indicating
+// if the resource that will be removed from the state should be destroyed too or not.
+type DestroyableConfigRemovable struct {
```

Comment:
It's a little weird having this in the addrs package.  It's not an address, it's a object that has an address and some associated configuration.

Reply:
Thanks. Really good point. Reworked this in 6ddecbbb2289be4367c888901eec4ca30c5e5452